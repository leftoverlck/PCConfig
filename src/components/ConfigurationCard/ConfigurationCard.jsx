import React from 'react';
import styles from './ConfigurationCard.module.css'; // Імпортуємо стилі

const ConfigurationCard = ({ configuration }) => {
    return (
        <div className={styles.configurationCard}>
            <h2 className={styles.configurationTitle}>Ваша конфігурація</h2>
            <div className={styles.specs}>
                <p>
                    <strong>Живлення (PSU):</strong> {configuration.psu}
                </p>
                <p>
                    <strong>Оперативна пам'ять (RAM):</strong> {configuration.ram}
                </p>
                <p>
                    <strong>Відеокарта (GPU):</strong> {configuration.gpu}
                </p>
                <p>
                    <strong>Процесор (CPU):</strong> {configuration.cpu}
                </p>
                <p>
                    <strong>Тип зберігання (Storage):</strong> {configuration.storage}
                </p>
                <p>
                    <strong>Материнська плата:</strong> {configuration.motherboard}
                </p>
            </div>
            <p className={styles.totalPrice}>
                <strong>Загальна вартість:</strong> {configuration.totalPrice}{' '}
            </p>
        </div>
    );
};

export default ConfigurationCard;
