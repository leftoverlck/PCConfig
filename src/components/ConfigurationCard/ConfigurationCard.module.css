/* ConfigurationCard.module.css */
.configurationCard {
    background-color: #d8d9d9;
    padding: 20px;
    border-radius: 8px;
    box-shadow: 0 4px 8px
}  