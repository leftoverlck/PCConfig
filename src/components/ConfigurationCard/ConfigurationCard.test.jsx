import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ConfigurationCard from './ConfigurationCard';

describe('ConfigurationCard', () => {
    const mockConfiguration = {
        psu: '750W',
        ram: '16GB',
        gpu: 'NVIDIA RTX 3080',
        cpu: 'Intel i9-11900K',
        storage: '1TB SSD',
        motherboard: 'ASUS ROG Strix Z590-E',
        totalPrice: '2500$',
    };

    test('renders the component with given configuration', () => {
        render(<ConfigurationCard configuration={mockConfiguration} />);
        
        expect(screen.getByRole('heading', { name: /Ваша конфігурація/i })).toBeInTheDocument();
        expect(screen.getByText(/750W/i)).toBeInTheDocument();
        expect(screen.getByText(/16GB/i)).toBeInTheDocument();
        expect(screen.getByText(/NVIDIA RTX 3080/i)).toBeInTheDocument();
        expect(screen.getByText(/Intel i9-11900K/i)).toBeInTheDocument();
        expect(screen.getByText(/1TB SSD/i)).toBeInTheDocument();
        expect(screen.getByText(/ASUS ROG Strix Z590-E/i)).toBeInTheDocument();
        expect(screen.getByText(/2500\$/i)).toBeInTheDocument();
    });
});
