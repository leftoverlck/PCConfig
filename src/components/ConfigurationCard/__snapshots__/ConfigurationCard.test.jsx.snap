// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigurationCard should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="configurationCard"
  >
    <h2
      class="configurationTitle"
    >
      Ваша конфігурація
    </h2>
    <div
      class="specs"
    >
      <p>
        <strong>
          Живлення (PSU):
        </strong>
         750W
      </p>
      <p>
        <strong>
          Оперативна пам'ять (RAM):
        </strong>
         16GB
      </p>
      <p>
        <strong>
          Відеокарта (GPU):
        </strong>
         NVIDIA RTX 3080
      </p>
      <p>
        <strong>
          Процесор (CPU):
        </strong>
         Intel i9-11900K
      </p>
      <p>
        <strong>
          Тип зберігання (Storage):
        </strong>
         1TB SSD
      </p>
      <p>
        <strong>
          Материнська плата:
        </strong>
         ASUS ROG Strix Z590-E
      </p>
    </div>
    <p
      class="totalPrice"
    >
      <strong>
        Загальна вартість:
      </strong>
       2500$ 
    </p>
  </div>
</DocumentFragment>
`;
